app/scripts/controllers/newOverview.js

```diff
@@ -0,0 +1,1200 @@
+  //   value: alerts object
+  var updatePodWarningsForObject = function(apiObject) {
+    var uid = getUID(apiObject);
+    if (!uid) {
+      return;
+    }
+
+    var pods = getPods(apiObject);
+    var notifications = ResourceAlertsService.getPodAlerts(pods, $routeParams.project);
+    setNotifications(apiObject, notifications);
+  };
+
+  // Updates pod warnings for a collection of API objects such as replication
+  // controllers or monopods.
+  var updatePodWarnings = function(apiObjects) {
+    _.each(apiObjects, updatePodWarningsForObject);
+  };
+
+  // Get the most recently-created replication controller for a deployment
+  // config. This might not be the active deployment if it was a failed or
+  // cancelled.
+  var getMostRecentReplicationController = function(deploymentConfig) {
+    var name = getName(deploymentConfig);
+    if (!name) {
+      return null;
+    }
+
+    return mostRecentByDeploymentConfig[name];
+  };
+
+  // Get the replication controllers that are displayed for a deployment
+  // config. This will return only the active replication controller unless a
+  // deployment is in progress.
+  var getVisibleReplicationControllers = function(deploymentConfig) {
+    var name = getName(deploymentConfig);
+    if (!name) {
+      return [];
+    }
+    return _.get(overview, ['replicationControllersByDeploymentConfig', name]);
+  };
+
+  // Get the "previous" replication controller (when a deployment is in
+  // progress and more than one is active). This is the donut that appears on
+  // the left when we show two. Returns null if there is no deployment in
+  // progress or the previous has been scaled down.
+  overview.getPreviousReplicationController = function(deploymentConfig) {
+    var replicationControllers = getVisibleReplicationControllers(deploymentConfig);
+    if (_.size(replicationControllers) < 2) {
+      return null;
+    }
+
+    // The array is sorted by age, most recent first. Return the second item.
+    return replicationControllers[1];
+  };
+
+  // Set warnings for a deployment config, including warnings for any active
+  // replication controllers and cancelled and failed deployments.
+  //
+  // Updates `state.notificationsByObjectUID`
+  //   key: object UID
+  //   value: alerts object
+  var updateDeploymentConfigWarnings = function(deploymentConfig) {
+    var notifications = {};
+
+    // Add any failed / canceled deployment notifications.
+    var mostRecent = getMostRecentReplicationController(deploymentConfig);
+    notifications = ResourceAlertsService.getDeploymentStatusAlerts(deploymentConfig, mostRecent);
+
+    // Roll up notifications like pod warnings for any visible replication controller.
+    var visibleReplicationControllers = getVisibleReplicationControllers(deploymentConfig);
+    _.each(visibleReplicationControllers, function(replicationController) {
+      var rcNotifications = getNotifications(replicationController);
+      _.assign(notifications, rcNotifications);
```

Comment:
The alert ID for RC notifications contains the object UID. (And for pods, we have the grouping logic.)